components/scream/src/share/property_checks/mass_and_energy_column_conservation_check.cpp

```diff
@@ -0,0 +1,331 @@
+  }
+  res_and_msg.msg = msg.str();
+
+  return res_and_msg;
+}
+
+
+KOKKOS_INLINE_FUNCTION
+Real MassAndEnergyColumnConservationCheck::compute_total_mass_on_column (const KT::MemberType&       team,
+                                                                         const uview_1d<const Real>& pseudo_density,
+                                                                         const uview_1d<const Real>& qv,
+                                                                         const uview_1d<const Real>& qc,
+                                                                         const uview_1d<const Real>& qi,
+                                                                         const uview_1d<const Real>& qr) const
+{
+  using PC = scream::physics::Constants<Real>;
+  const Real gravit = PC::gravit;
+
+  Real total_mass(0);
+  ekat::ExeSpaceUtils<typename KT::ExeSpace>::parallel_reduce(team, 0, m_num_levs,
+                                                              [&] (const int lev, Real& local_mass) {
+    local_mass += (qv(lev)+
+                   qc(lev)+
+                   qi(lev)+
+                   qr(lev))*pseudo_density(lev)/gravit;
+  }, total_mass);
+
+  return total_mass;
+}
+
+KOKKOS_INLINE_FUNCTION
+Real MassAndEnergyColumnConservationCheck::compute_mass_boundary_flux_on_column (const Real vapor_flux,
+                                                                                 const Real water_flux) const
+{
+  using PC = scream::physics::Constants<Real>;
+  const Real RHO_H2O  = PC::RHO_H2O;
+
+  return vapor_flux - water_flux*RHO_H2O;
+}
+
+KOKKOS_INLINE_FUNCTION
+Real MassAndEnergyColumnConservationCheck::compute_total_energy_on_column (const KT::MemberType&       team,
+                                                                           const uview_1d<const Real>& pseudo_density,
+                                                                           const uview_1d<const Real>& T_mid,
+                                                                           const uview_2d<const Real>& horiz_winds,
+                                                                           const uview_1d<const Real>& qv,
+                                                                           const uview_1d<const Real>& qc,
+                                                                           const uview_1d<const Real>& qr,
+                                                                           const Real                  ps,
+                                                                           const Real                  phis) const
+{
+  using PC = scream::physics::Constants<Real>;
+  const Real LatVap = PC::LatVap;
+  const Real LatIce = PC::LatIce;
+  const Real gravit = PC::gravit;
+  const Real Cpair  = PC::Cpair;
+
+  Real total_energy(0);
+  ekat::ExeSpaceUtils<typename KT::ExeSpace>::parallel_reduce(team, 0, m_num_levs,
```

Comment:
Super minor (and perhaps subjective), but in case you're going to add a commit with other mods, you might as well consider adding a class-level typedef for this type, and then use it in all the fcns in the cpp file to keep lines shorter and more readable (I see `ESU` a short name in other areas of the code).